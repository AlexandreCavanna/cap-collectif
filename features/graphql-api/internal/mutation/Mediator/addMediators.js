/* eslint-env jest */
import '../../../_setup';


const AddMediatorsMutation = /* GraphQL*/ `
  mutation AddMediators($input: AddMediatorsInput!) {
    addMediators(input: $input) {
      step {
        __typename
        title
      }
      mediators {
        edges {
          node {
            user {
              username
              roles
            }
          }
        }
      }
    }
  }
`

describe('mutations|addMediators', () => {
  beforeEach(async () => {
    await global.enableFeatureFlag('mediator');
  });
  it('admin add users as mediator.', async () => {

    const variables = {
      "input": {
        "usersId": [
          "VXNlcjp1c2VyNg==",
          "VXNlcjp1c2VyNw=="
        ],
        "stepId": "U2VsZWN0aW9uU3RlcDpzZWxlY3Rpb25zdGVwMg=="
      }
    }
    const addMediators = await graphql(
      AddMediatorsMutation,
      variables,
      'internal_admin',
    );
    expect(addMediators).toMatchSnapshot();
  });
});
