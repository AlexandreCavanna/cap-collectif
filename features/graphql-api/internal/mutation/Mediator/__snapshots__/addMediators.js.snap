// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations|addMediators admin add users as mediator. 1`] = `
Object {
  "addMediators": Object {
    "mediators": Object {
      "edges": Array [
        Object {
          "node": Object {
            "user": Object {
              "roles": Array [
                "ROLE_USER",
                "ROLE_MEDIATOR",
              ],
              "username": "user6",
            },
          },
        },
        Object {
          "node": Object {
            "user": Object {
              "roles": Array [
                "ROLE_USER",
                "ROLE_MEDIATOR",
              ],
              "username": "user7",
            },
          },
        },
      ],
    },
    "step": Object {
      "__typename": "SelectionStep",
      "title": "Vainqueur",
    },
  },
}
`;
